baseplate/frameworks/pyramid/__init__.py

```diff
@@ -306,10 +365,22 @@ def _on_application_created(self, event: pyramid.events.ApplicationCreated) -> N
 
     def _on_new_request(self, event: pyramid.events.ContextFound) -> None:
         request = event.request
+        endpoint = ""
+
+        if request.matched_route:
+            endpoint = (
+                request.matched_route.pattern
+                if (hasattr(request.matched_route, "pattern") and request.matched_route.pattern)
+                else request.matched_route.name
+            )
+        else:
+            endpoint = "404"
+        request.prom_metrics_enabled = True
+        request.start_time = time.perf_counter()
```

Comment:
I'm not entirely a fan of starting to put our own fields inside of the request field. But if we're going to do that I would do two things:
- give them very reddit names (baseplate_prometheus_start_time or smth)
- why not put the endpoint in there as well?

Reply:
I agree, not a huge fan either, but I think it's better and easier than most of the alternatives since the request is the thing that's consistently passed between funcs

> give them very reddit names
because they're not specific to reddit, which does open us more to collisions but I'm not really concerned about that. how about a `_reddit_...` prefix on them so it's not too long, since the data itself isn't actually specific to prom for the start time?

> put the endpoint in there
it already is via request.matched_route

Reply:
> it already is via request.matched_route
well except we have this string of 3 different logic tests because we can't trust it's there

That's why I was saying that :-)

Reply:
> because they're not specific to reddit, which does open us more to collisions but I'm not really concerned about that.
Well you threw me for a loop when I read `request.start_time` because I didn't understand where it was coming from at first.  
If it had an obnoxiously long name related to reddit/baseplate/prometheus it would have been more obvious is all. Also yes name collision :-)

Reply:
prefixed with reddit_ and tried to re-use the endpoint where possible